tests/calculators/test_aims.py

```diff
@@ -0,0 +1,166 @@
+import os
+import pytest
+
+import numpy as np
+from ase import Atoms
+from ase.build import bulk
+
+import wfl.calculators.aims
+from wfl.calculators import generic
+from wfl.configset import OutputSpec
+
+
+skipif_condition = 'ASE_AIMS_COMMAND' not in os.environ or 'AIMS_SPECIES_DIR' not in os.environ \
+                   or not os.environ['AIMS_SPECIES_DIR'].endswith('light') \
+                   or 'OMP_NUM_THREADS' not in os.environ or os.environ['OMP_NUM_THREADS'] != "1"
+skipif_reason = 'Missing env var ASE_AIMS_COMMAND or ASE_SPECIES_DIR or ' + \
+                'ASE_SPECIES_DIR does not refer to light-settings or ' + \
+                'OMP_NUM_THREADS or OMP_NUM_THREADS has not correctly been set to 1'
+
+
+@pytest.fixture
+def parameters_nonperiodic():
+    parameters = {
+        'xc': 'pbe',
+        'spin': 'none',
+        'relativistic': 'none',
+        'charge': 0.,
+        'sc_iter_limit': 500,
+        'occupation_type': 'gaussian 0.01',
+        'charge_mix_param': 0.6,
+        'mixer': 'pulay',
+        'n_max_pulay': 10,
+        'sc_accuracy_rho': 1e-2,
+        'sc_accuracy_eev': 1e-2,
+        'sc_accuracy_etot': 1e-4,
+        'sc_accuracy_forces': 1E-2,
+        'compute_forces': True,
+        'KS_method': 'parallel',
+    }
+    return parameters
+
+
+def test_setup_calc_params(parameters_nonperiodic):
+
+    parameters = parameters_nonperiodic
+    parameters_periodic = {
+        'k_grid': '1 1 1',
+        'k_grid_density': 0.1,
+        'k_offset': 0.1,
+        'relax_unit_cell': 'full',
+        'external_pressure': 10,
+        'compute_analytical_stress': '.true.',
```

Comment:
Having settings like `compute_analytical_stress` required to be set correctly outside the calculator makes it impossible to cleanly apply the workflow to a combination of periodic and non-periodic configurations, e.g. what you might end up with when you are reevaluating an entire database which includes crystal cells as well as isolated atoms. 

Could we have this dealt with automatically inside the calculator, which should know both whether a configuration is periodic (`ase.Atoms.pbc`) and whether a stress has been requested from that calculation (if it's even slow enough to be worth turning off)

Reply:
The current version handles this automatically inside the calculator in some sense. For each structure it detects if its periodic or non-periodic. If its periodic it starts the calculation with all the (FHI-Aims) settings passed. If its non-periodic, it removes all of the additional key-value pairs that need to be passed for a periodic calculation.

Reply:
OK - I guess if the workflow removes invalid keywords it's OK.  Maybe we should document in the calculator's docstring what keywords get removed when pbc is False.

Reply:
Agreed. I'll adjust the docstring.

Related to this: What is your opinion on this alternative that will allow the user specifying individual settings for each configuration in a database?
`generic.run()` gets an additional argument `calculator_settings` which is a list of dictionaries (each dictionary representing the `kwargs` used to define the Calculator in the form `Calculator = (initializer, args, kwargs)`, e.g. one `kwargs`-setting for PBC, one for non-PBC, one for isolated atoms or so). When looping over the configurations in the users database, the calculator gets constructed individually for each config (i.e. shifting the `construct_calculator_picklesafe()` inside the loop) where the required `kwargs` gets selected from the list in `calculator_settings` based on an index that needs to be specified in the Atoms.info for each config (something like `Atoms.info['calculator_settings_index'] = 1`).

Reply:
I thought about this before, although not with a short list and an index, but zipping the configs and a list of calculators, one per config.

What exactly is the use case? If it's just periodic vs. non-periodic, I think I prefer that being handled automatically in the calculator, like we've done so far. Do you have a specific example where that can't be done cleanly, or another use for this feature? 

The only one I can thin think of is to be able to explicitly specify k-point meshes without needing to rely on k-spacing, which can have problems with small strain causing a discontinuous jump in the precise number along particular directions.  It's pretty unusual, though, and yet still perhaps sufficiently ubiquitous across calculators that I' might prefer a special `Atoms.info["DFT_k_pt_mesh"]` field which all the wfl DFT calculators know about, so users don't have to change anything when changing DFT calculators.

Reply:
> What exactly is the use case? If it's just periodic vs. non-periodic, I think I prefer that being handled automatically in the calculator, like we've done so far. Do you have a specific example where that can't be done cleanly, or another use for this feature?

The first thing I was thinking about was that the isolated atoms can sometimes be tricky and need some more specific setting. Other situations might be that a user wants to set a keyword such that the geometry gets optimized, but only for a certain part of the structures in the database, or a user is interested in differences (e.g. different VdW-corrections or so) for the structures in a database.

Reply:
I think if it's just two groups (isolated vs. everything else), I think it's not worth the complexity - you can just do two separate evaluation calls, and the isolated atoms are usually pretty cheap anyway.

I'm also down on using the DFT calculator to do the relaxation, although others may like that.  I prefer to use our minimizers, whose behavior I understand, with the calculator just providing gradients.  I do admit it's typically less efficient, though.

Reply:
I would also lean towards less complexity unless there's clear demand for the functionality or significant improvement. 

As for DFT relaxations, it was 10 times more efficient to do it directly with ORCA, so I explicitly use that. Wfl ORCA wrapper has that implemented.

Reply:
that was number of geometry optimisation steps. ORCA is using internal coordinates and I would imagine it can reuse some of the electronic structure information from the previous geometry opt step. I wonder if internal using internal coordinates is a bit less beneficial for materials than it is for organic molecules.